lib/publiccloud/instance.pm

```diff
@@ -223,9 +228,35 @@ sub wait_for_ssh
         sleep 1;
     }
 
-    croak(sprintf("Unable to reach SSH port of instance %s with public IP:%s within %d seconds",
-            $self->{instance_id}, $self->{public_ip}, $args{timeout}))
-      unless ($args{proceed_on_failure});
+    # We already considering waiting as failed but sometimes it make sense to wait a bit longer to see that instance become available after all
+    my $duration = 0;
+    while (($duration = time() - $start_time) < $args{hardtimeout}) {
+        unless ($port_reachable) {
+            if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0) {
+                $port_reachable = 1;
+            }
+        }
+        if ($self->run_ssh_command(cmd => 'sudo journalctl -b | grep -E "Reached target (Cloud-init|Default|Main User Target)"', proceed_on_failure => 1, quiet => 1, username => $args{username}) =~ m/Reached target.*/) {
+            $delayed_success = 1;
```

Comment:
If we reach this point, then you can also set `$port_reachable = 1`. Also, I guess here a `last` is missing, otherwise it loops until a timeout happens.